firebase_admin/mlkit.py

```diff
@@ -459,6 +550,85 @@ def __init__(self, app):
+                (None for no limit)
+            max_time_seconds: The maximum seconds to try polling for operation complete.
+                (None for no limit)
+            always_return_model: If true, returns a locked Model instead of raising deadline
+                exceeded exceptions.
+
+        Returns:
+            dict: A dictionary of the returned model properties.
+
+        Raises:
+            TypeError: if the operation is not a dictionary.
+            ValueError: If the operation is malformed.
+        """
+        if not isinstance(operation, dict):
+            raise TypeError('Operation must be a dictionary.')
+        op_name = operation.get('name')
+        _, model_id = _validate_and_parse_operation_name(op_name)
+
+        current_attempt = 0
+        start_time = datetime.datetime.now()
+        stop_time = (None if max_time_seconds is None else
+                     start_time + datetime.timedelta(seconds=max_time_seconds))
+        while True:
+            if operation.get('done'):
+                if operation.get('response'):
+                    return operation.get('response')
+                elif operation.get('error'):
+                    raise _utils.handle_operation_error(operation.get('error'))
+                else:
+                    # A 'done' operation must have either a response or an error.
+                    raise ValueError('Operation is malformed.')
+            else:
+                # We just got this operation. Wait before getting another
+                # so we don't exceed the GetOperation maximum request rate.
+                if max_polling_attempts is not None and current_attempt >= max_polling_attempts:
+                    if always_return_model:
+                        return get_model(model_id).as_dict()
+                    raise exceptions.DeadlineExceededError('Polling max attempts exceeded.')
+                delay_factor = pow(
+                    _MLKitService.POLL_EXPONENTIAL_BACKOFF_FACTOR, current_attempt)
+                wait_time_seconds = delay_factor * _MLKitService.POLL_BASE_WAIT_TIME_SECONDS
+                after_sleep_time = (datetime.datetime.now() +
+                                    datetime.timedelta(seconds=wait_time_seconds))
+                if stop_time is not None and after_sleep_time > stop_time:
+                    if always_return_model:
+                        return get_model(model_id).as_dict()
+                    raise exceptions.DeadlineExceededError('Polling max time exceeded.')
+                time.sleep(wait_time_seconds)
+                operation = self.get_operation(op_name)
+                current_attempt += 1
+
+
+    def create_model(self, model):
+        _validate_model(model)
+        try:
+            return self.handle_operation(
+                self._client.body('post', url='models', json=model.as_dict()),
+                max_polling_attempts=1,
```

Comment:
I think this shouldn't poll at all. It should be `max_polling_attempts=0`.

Reply:
I can do that, but in practice almost nothing is done on the initial operation return from the create/update functions, but a lot of faster things are done on the first poll. That's why I set it to poll once. 
Why do you prefer no polling?